test/ability_cloud_nine.c

```diff
@@ -0,0 +1,16 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Cloud Nine prevents weather effects")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_SANDSTORM].effect == EFFECT_SANDSTORM);
+        PLAYER(SPECIES_PSYDUCK) { Ability(ABILITY_CLOUD_NINE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SANDSTORM); }
+        TURN {}
+    } SCENE {
+        NONE_OF { HP_BAR(player); }
```

Comment:
You might like to check that the sandstorm move and/or end-of-turn animation doesn't play?

Reply:
I think if I'm reading the ability correctly on bulbapedia the animations should still play (and they do) - the effects are just negated.